sandstone/pattern-video-player/src/App/App.js

```diff
@@ -16,107 +18,95 @@ import css from './App.module.less';
-
-	handleSelectBreadcrumb = ({index}) => this.setState({panelIndex: index});
-
-	handleHidePanelsClick = () => this.setState({panelsVisible: false});
-
-	handleShowPanelsClick = () => {
-		this.videoRef.hideControls();
-		this.setState({panelsVisible: true});
+	}, [panelsVisible]);
+
+	let videoRef;
+	const handleNextPanelClick = () => setPanelIndex(prevPanelIndex => (prevPanelIndex + 1));
+	const handleSelectBreadcrumb = ({index}) => setPanelIndex(index);
+	const handleHidePanelsClick = () => setPanelsVisible(false);
+	const handleShowPanelsClick = () => {
+		videoRef.hideControls();
+		setPanelsVisible(true);
 	};
 
-	setVideoIndex = (videoIndex) => this.setState({videoIndex});
-
-	setVideoRef = (ref) => {
-		this.videoRef = ref;
+	const handelVideoIndexChange = (index) => setVideoIndex(index);
+	const setVideoRef = (ref) => {
+		videoRef = ref;
 	};
 
-	render () {
-		const {className, ...rest} = this.props;
-		const {source, desc, ...restVideo} = getVideo(this.state.videoIndex);
-		delete rest.panelIndex;
-		delete rest.videoIndex;
-		return (
-			<div {...rest} className={className + ' ' + css.app}>
-				<VideoPlayer {...restVideo} className={css.player + ' enact-fit'} ref={this.setVideoRef} spotlightDisabled={this.state.panelsVisible}>
-					<source src={source} type="video/mp4" />
-					<infoComponents>
-						{desc}
-					</infoComponents>
-					<MediaControls actionGuideLabel="Press Down Button">
-						<Button
-							icon="list"
-							backgroundOpacity="transparent"
-							onClick={this.handleShowPanelsClick}
-							spotlightDisabled={this.state.panelsVisible}
-						/>
-					</MediaControls>
-				</VideoPlayer>
-				{this.state.panelsVisible ?
-					<Panels
-						index={this.state.panelIndex}
-						onBack={this.handleSelectBreadcrumb}
-					>
-						<MainPanel
-							onHidePanels={this.handleHidePanelsClick}
-							onNextPanel={this.handleNextPanelClick}
-							onVideoIndexChange={this.setVideoIndex}
-							spotlightId="main-panel"
-							title="Videos"
-							videoIndex={this.state.videoIndex}
-						/>
-						<ItemPanel title="Second" />
-					</Panels> :
-					null}
-			</div>
-		);
-	}
-}
+	const {className, ...rest} = props;
+	const {source, desc, ...restVideo} = getVideo(videoIndex);
+	delete rest.panelIndex;
+	delete rest.videoIndex;
+	return (
+		<div {...rest} className={className + ' ' + css.app}>
+			<VideoPlayer {...restVideo} className={css.player + ' enact-fit'} ref={setVideoRef} spotlightDisabled={panelsVisible}>
+				<source src={source} type="video/mp4" />
+				<infoComponents>
+					{desc}
+				</infoComponents>
+				<MediaControls actionGuideLabel="Press Down Button">
+					<Button
+						icon="list"
+						backgroundOpacity="transparent"
+						onClick={handleShowPanelsClick}
+						spotlightDisabled={panelsVisible}
+					/>
+				</MediaControls>
+			</VideoPlayer>
+			{panelsVisible ?
+				<Panels
+					index={panelIndex}
+					onBack={handleSelectBreadcrumb}
```

Comment:
Let's rename it since it's sandstone.

```suggestion
					onBack={handleBack}
```

Reply:
fixed it.